src/catalog/src/durable/impls/persist.rs

```diff
@@ -664,6 +680,26 @@ impl PersistCatalogState {
 
         self.apply_updates(updates)?;
 
+        // Advance the since, allowing Persist to compact the data.
+        // Lag the shard's upper by 1 to keep it readable.
+        // Since nothing relies on this since being tight,
+        // we use the `maybe_` variant and do not fence.
+        let downgrade_to = Antichain::from_elem(self.upper.saturating_sub(1));
+        let opaque = i64::initial();
+        let downgrade = self
+            .since_handle
+            .maybe_compare_and_downgrade_since(&opaque, (&opaque, &downgrade_to))
+            .await;
```

Comment:
What is `opaque` for?

Reply:
We have a little doc on this: https://github.com/MaterializeInc/materialize/blob/bdc4726f57269b0e857e2e1823eabfab13fafa1a/src/persist-client/src/critical.rs#L273-L278

I don't think we actually care about fencing in this case, so this effectively ignores it by always using the same value as the token so the comparison never fails.